src/android/org/jboss/aerogear/cordova/push/PushPlugin.java

```diff
@@ -230,7 +229,7 @@ private PushRegistrar getPushRegistrar() {
    */
   public static void sendMessage(Bundle message) {
     if (message != null) {
-      if (sendMetrics && !foreground) {
+      if (sendMetrics && (!foreground || cachedMessage != null)) {
```

Comment:
It's explained in the description of the PR. When the app is fully closed and you open it from a push, first time the message is cached but not processed, and when the plugin is ready and the app calls register again, then checks if there is a cached message and process it, but the app is in foreground, so the received confirmation is not sent. With this check we can know if the app was launched from a push message.